### ISSUE TITLE
Regularly check for `PreviousStepStillProcessingError` cache entries

### ISSUE THREAD

**Issue by severo** [huggingface/datasets-server#2099]
`PreviousStepStillProcessingError` errors should be transient cache entries. If they exist for longer than... days, we should consider them obsolete and refresh their ancestors, or, simpler: the whole dataset.

**Comment by severo** [comment]
It's a follow-up of https://github.com/huggingface/datasets-server/pull/2093. First, let's wait for some days/weeks and look at the cache collection to see if it's a real issue.

For example: to get the list of `PreviousStepStillProcessingError` cache entries older than 3 days:

```
use datasets_server_cache
db.cachedResponsesBlue.countDocuments({error_code: "PreviousStepStillProcessingError", updated_at: {$lt: new Date((new Date())-1000*60*60*24*3)}})
```

**Comment by severo** [comment]
```js
// More than 3 days
db.cachedResponsesBlue.countDocuments({error_code: "PreviousStepStillProcessingError", updated_at: {$lt: new Date((new Date())-1000*60*60*24*3)}})
103
// More than 1 month
db.cachedResponsesBlue.countDocuments({error_code: "PreviousStepStillProcessingError", updated_at: {$lt: new Date((new Date())-1000*60*60*24*30)}})
63
```

**Comment by severo** [comment]
fixed by #2491
